escrow/payment_handler.go

```diff
@@ -93,7 +96,39 @@ func (h *paymentChannelPaymentHandler) getPaymentFromContext(context *handler.Gr
 }
 
 func (h *paymentChannelPaymentHandler) Complete(payment handler.Payment) (err *handler.GrpcError) {
-	return paymentErrorToGrpcError(payment.(*paymentTransaction).Commit())
+	if err = paymentErrorToGrpcError(payment.(*paymentTransaction).Commit()); err != nil {
+		PublishChannelStats(payment)
+	}
+	return err
+}
+
+func PublishChannelStats(payment handler.Payment) (err *handler.GrpcError) {
+	if !config.GetBool(config.MeteringEnabled)  {
+		err = handler.NewGrpcErrorf(codes.Internal, "Cannot post latest offline channel state as metering is disabled !!")
+		log.WithError(err.Err())
```

Comment:
I guess `WithError` will not log anything, though I am not sure, but I would add a message:
```suggestion
		log.WithError(err.Err()).Error("Error in payment handler commit")
```